packages/zowe-explorer-api/src/vscode/ui/TableView.ts

```diff
@@ -0,0 +1,571 @@
+        }
+
+        /**
+         * Add one or more actions to the given row.
+         *
+         * @param index The row index where the action should be displayed
+         * @param actions The actions to add to the given row
+         *
+         * @returns Whether the webview successfully received the new action(s)
+         */
+        public addAction(index: number | "all", ...actions: ActionOpts[]): Promise<boolean> {
+            if (this.data.actions[index]) {
+                const existingActions = this.data.actions[index];
+                this.data.actions[index] = [...existingActions, ...actions.map((action) => ({ ...action, condition: action.condition?.toString() }))];
+            } else {
+                this.data.actions[index] = actions.map((action) => ({ ...action, condition: action.condition?.toString() }));
+            }
+            return this.updateWebview();
+        }
+
+        /**
+         * Add one or more context menu options to the given row.
+         *
+         * @param id The row index or column ID where the action should be displayed
+         * @param actions The actions to add to the given row
+         * @returns Whether the webview successfully received the new context menu option(s)
+         */
+        public addContextOption(id: number | "all", ...options: ContextMenuOpts[]): Promise<boolean> {
+            if (this.data.contextOpts[id]) {
+                const existingOpts = this.data.contextOpts[id];
+                this.data.contextOpts[id] = [...existingOpts, ...options.map((option) => ({ ...option, condition: option.condition?.toString() }))];
+            } else {
+                this.data.contextOpts[id] = options.map((option) => ({ ...option, condition: option.condition?.toString() }));
+            }
+            return this.updateWebview();
+        }
+
+        /**
+         * Get rows of content from the table view.
+         * @param rows The rows of data in the table
+         * @returns Whether the webview successfully received the new content
+         */
+        public getContent(): RowData[] {
+            return this.data.rows;
+        }
+
+        /**
+         * Add rows of content to the table view.
+         * @param rows The rows of data to add to the table
+         * @returns Whether the webview successfully received the new content
+         */
+        public async addContent(...rows: RowData[]): Promise<boolean> {
+            this.data.rows.push(...rows);
+            return this.updateWebview();
+        }
+
+        /**
+         * Update an existing row in the table view.
+         * @param index The AG GRID row index to update within the table
+         * @param row The new row content
+         * @returns Whether the webview successfully updated the new row
+         */
+        public async updateRow(index: number, row: RowData | null): Promise<boolean> {
```

Comment:
Should we mention in the Typedoc for this method, that if `row` is null then the row will be deleted?

Reply:
Yes, good call, thanks - addressed in upcoming commit 👍